[libSyntax] Model parameter modifiers as a `ModifierList`

ahoppen opened the pull request:
This fixes a libSyntax misparse where the first parameter name was parsed as the `isolated` token.

ahoppen commented:
@swift-ci Please smoke test

rintaro commented:
I'm not sure this is right. Can  `isolated` and `_const` be used at the same time?
```
func foo(isolated _const _ arg: SomeTy) { }
```
The current parser emits an error `error: 'isolated' may only be used on parameters`, but I don't think this is a correct error message.
If we can use both, should they be ordered? As far as I can see in the C++ parser code, they are not ordered (`_const isolated` is allowed). If so, this portion should be modeled as a list, like `ModifierList`.
If we can only use a modifier on a type, this part should be modeled as like:
```python
             Child('Specifier', kind='ContextualKeywordToken',
                   is_optional=True),
```
Either way, we should remap this to `tok::contextual_keyword`.

rintaro commented:
Considering the possibility of new modifiers being introduced in the future, I'd suggest modeling it as a `ModifierList` which is consistent with other value decls.
```python
    Node('FunctionParameter', name_for_diagnostics='function parameter', kind='Syntax',
         traits=['WithTrailingComma'],
         children=[
             Child('Attributes', kind='AttributeList', name_for_diagnostics='attributes',
                   collection_element_name='Attribute', is_optional=True),
             Child('Modifiers', kind='ModifierList', name_for_diagnostics='modifiers',
                   collection_element_name='Modifier', is_optional=True),
             Child('FirstName', kind='Token', name_for_diagnostics='name',
                   ...
```

ahoppen committed:
[libSyntax] Model parameter modifiers as a `ModifierList`

This fixes a libSyntax misparse where the first parameter name was parsed as the `isolated` token.

ahoppen commented:
https://github.com/apple/swift-syntax/pull/861

@swift-ci Please smoke test

ahoppen commented:
https://github.com/apple/swift-syntax/pull/861
https://github.com/apple/swift-stress-tester/pull/193

@swift-ci Please smoke test

rintaro commented:
https://github.com/apple/swift-syntax/pull/861
https://github.com/apple/swift-stress-tester/pull/193

@swift-ci Please smoke test Linux

ahoppen commented:
https://github.com/apple/swift-syntax/pull/861
https://github.com/apple/swift-stress-tester/pull/193

@swift-ci Please smoke test Linux
